.github/workflows/integration-test.yml

```diff
@@ -498,6 +498,7 @@ jobs:
             terraform init
             if terraform apply --auto-approve \
               -var="ssh_key_value=${PRIVATE_KEY}" -var="github_test_repo=${{env.CWA_GITHUB_TEST_REPO_URL}}" \
+              -var="test_name=${{ matrix.arrays.os }}" \
```

Comment:
Does `matrix.arrays.os` store the operating system?
I am confused why it is being assigned into another variable called `test_name`?

Reply:
`matrix.arrays.os` stores the OS that the test is running on. It is assigned to test_name because that is the variable name being used to distinguish the instance names for different OS. For example looking at `main.tf` file for mac in [amazon-cloudwatch-agent-test repo](https://github.com/aws/amazon-cloudwatch-agent-test/blob/88b1a960e56528436473416a7040b06375917e64/terraform/ec2/mac/main.tf#L70), it's the variable being used to dynamically name the instances.

We could add a variable called `test_os` instead for clarity but it accomplishes the same thing.

Reply:
In your example, we already differentiate using the prefix `cwagent-integ-test-ec2-mac-` which includes the OS it's running on. There's no harm in adding this, but I'm not sure I understand why it should be the `matrix.array.os`.

Reply:
Adding `test_name` to `cwagent-integ-test-ec2-mac-` will also specify which major version of Mac we are using, since it's referring to the os in [this file](https://github.com/aws/amazon-cloudwatch-agent-test/blob/main/generator/resources/ec2_mac_test_matrix.json).

As for why it should be `matrix.array.os`, it's the only variable that is common to all test_matrix and has clear distinctions. It's also what is being used in the internal repo so I thought that makes the most sense.

Reply:
Technically for the main purpose of this PR, the test_name addition is not necessary, but it's just something I noticed during testing that all of our test instances end up with names such as `cwagent-integ-test-ec2---90a1bd2bdc0f9e70)` which makes it hard to pinpoint which stage of integration test we should look into since each stage is usually OS based.

Reply:
So instead it would be something like `cwagent-integ-test-ec2-mac-macOS Big Sur Amd64-[testing-id]`. Seems fine.

Reply:
That's a good point. I don't think EC2 tag allows space. In this case we can change the `os` variable value in the matrix file or just remove `test_name` and leave it as it is for Mac. I'm fine with either option.